push-build.sh

```diff
@@ -185,8 +186,19 @@ if ((FLAGS_federation)); then
   common::stepheader PUSH FEDERATION
   ############################################################################
   logecho -n "Push federation images: "
-  # FEDERATION_PUSH_REPO_BASE should be set by the calling job (yaml)
-  logrun -s ${KUBE_ROOT}/federation/develop/push-federation-images.sh
+  GCLOUD_PROJECT=$($GCLOUD config list project --format 'value(core.project)')
+  if [[ -z "${GCLOUD_PROJECT}" ]]; then
+    logecho "No account authorized through gcloud. So using FEDERATION_PUSH_REPO_BASE to push images"
+    KUBE_DOCKER_REGISTRY=${FEDERATION_PUSH_REPO_BASE}
+  else
+    logecho "Account authorized through gcloud. So using GCLOUD_PROJECT to push images"
+    KUBE_DOCKER_REGISTRY="gcr.io/${GCLOUD_PROJECT}"
+  fi
+
+  release::docker::release \
```

Comment:
but this also pushes kube-apiserver, kube-controller-manager, kube-scheduler, and kube-proxy. do we want that?